### PR TITLE
move [Discourse] url to query param

### PR DESCRIPTION
**Pull request by calebcartwright**
Refs #3714

### CODE REVIEW

**Comment by calebcartwright** [comment]
Oops! Didn't realize that we were using the Discourse badge as part of the e2e test suite. Fixing that now

File: services/discourse/discourse.service.js [hunk 1]
```diff
@@ -13,6 +13,10 @@ const schema = Joi.object({
   like_count: nonNegativeInteger,
 }).required()
 
+const queryParamSchema = Joi.object({
+  hostUrl: optionalUrl.required(),
```
Comment (paulmelnikow) [inline services/discourse/discourse.service.js]: I don't feel strongly, though another option for naming this parameter might be `server`.

File: services/discourse/discourse.service.js [hunk 2]
```diff
@@ -21,20 +25,19 @@ class DiscourseBase extends BaseJsonService {
   static buildRoute(metric) {
     return {
       base: 'discourse',
-      // Do not base new services on this route pattern.
-      // See https://github.com/badges/shields/issues/3714
-      pattern: `:scheme(http|https)/:host/${metric}`,
+      pattern: `${metric}`,
```
Comment (paulmelnikow) [inline services/discourse/discourse.service.js]:
```suggestion
      pattern: metric,
```

File: services/discourse/discourse.service.js (hunk 1 above)
Comment (calebcartwright) [reply services/discourse/discourse.service.js]: I like that better 👍 I struggled a bit naming these, so I'm not attached to `hostUrl` either

File: services/discourse/discourse.service.js (hunk 2 above)
Comment (calebcartwright) [reply services/discourse/discourse.service.js]: Good spot! That's what I get for being lazy with the delete key 😉

**Pull request merged**